src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java

```diff
@@ -48,15 +49,22 @@
     @Argument(doc = "If specified, shard the output bam", shortName = "shardedOutput", fullName = "shardedOutput", optional = true)
     private boolean shardedOutput = false;
 
+    @Argument(doc="The output parallelism, sets the number of reducers. Defaults to one partition per 10MB of input.",
+            shortName = "P", fullName = "parallelism", optional = true)
+    protected int parallelism = 0;
+
     @Override
     protected void runTool(JavaSparkContext ctx) {
         JavaRDD<GATKRead> initialReads = getReads();
+        if (parallelism == 0) {
+            parallelism = ReadsSparkSource.getRecommendedParallelism(ctx, getReadSourceName(), initialReads.partitions().size());
+        }
```

Comment:
Added a method for this, `getParallelismPartitionSize()`, which can be overridden by subclasses.